ext/datasource/refreshable_file/file_datasource.go

```diff
@@ -0,0 +1,108 @@
+package refreshable_file
+
+import (
+	"github.com/alibaba/sentinel-golang/ext/datasource"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/alibaba/sentinel-golang/util"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+var (
+	logger = logging.GetDefaultLogger()
+)
+
+type RefreshableFileDataSource struct {
+	datasource.Base
+	sourceFilePath string
+	isInitialized  util.AtomicBool
+	watcher        *fsnotify.Watcher
+}
+
+func FileDataSourceStarter(sourceFilePath string, handlers ...datasource.PropertyHandler) (*RefreshableFileDataSource, error) {
+	var ds = &RefreshableFileDataSource{
+		sourceFilePath: sourceFilePath,
+	}
+	for _, h := range handlers {
+		ds.AddPropertyHandler(h)
+	}
+	return ds, ds.Initialize()
+}
+
+func (s *RefreshableFileDataSource) ReadSource() ([]byte, error) {
+	f, err := os.Open(s.sourceFilePath)
+	defer f.Close()
+
+	if err != nil {
+		return nil, errors.Errorf("The rules file is not existed, err: %+v.", errors.WithStack(err))
+	}
+	src, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, errors.Errorf("Fail to read file, err: %+v.", errors.WithStack(err))
+	}
+	return src, nil
+}
+
+func (s *RefreshableFileDataSource) Initialize() error {
+	if !s.isInitialized.CompareAndSet(false, true) {
+		return errors.New("RefreshableFileDataSource had been Initialized.")
+	}
+
+	s.doUpdate()
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		return errors.Errorf("Fail to new a w of fsnotify, err: %+v", err)
```

Comment:
watcher?

Reply:
yes, it's caused by rename, I will rename here.

Reply:
Yes, it's caused by mis rename. I will fix here.